tests/integration/new-integration-test.sh

```diff
@@ -0,0 +1,197 @@
+set -eo pipefail
+
+# This script is expected to be called from the root folder of Connaisseur
+declare -A DEPLOYMENT_VALIDITY=(["VALID"]="0" ["INVALID"]="0")
+RED="\033[0;31m"
+GREEN="\033[0;32m"
+NC="\033[0m"
+SUCCESS="${GREEN}SUCCESS${NC}"
+FAILED="${RED}FAILED${NC}"
+N=1
+EXIT="0"
+
+### SINGLE TEST CASE ####################################
+single_test() {  # ID TXT TYP IMG NS OUT VAL
+    echo -n "[$1] $2"
+    if [[ "$3" == "deploy" ]]; then
+        kubectl run pod-$1 --image="$4" --namespace="$5" >output.log 2>&1 || true
+    else
+        kubectl apply -f $4 >output.log 2>&1 || true
+    fi
+    if [[ ! "$(cat output.log)" =~ "$6" ]]; then
+        echo -e ${FAILED}
+        echo "::group::Output"
+        cat output.log
+        kubectl logs -n connaisseur -lapp.kubernetes.io/instance=connaisseur
+        echo "::endgroup::"
+        EXIT="1"
+    else
+        echo -e "${SUCCESS}"
+    fi
+
+    if [[ -n "$7" ]]; then
+      DEPLOYMENT_VALIDITY[$7]=$((${DEPLOYMENT_VALIDITY[$7]}+1))
+    fi
+}
+
+### MULTI TEST CASE FROM FILE ####################################
+multi_test() {
+  while read test_case
+  do
+    IFS=";" read -r -a arr <<< "${test_case}"
+    TEST_CASE_TXT=${arr[0]}
+    TYPE=${arr[1]}
+    IMAGE=${arr[2]}
+    NAMESPACE=${arr[3]}
+    OUTPUT=${arr[4]}
+    VALIDITY=${arr[5]}
+    single_test "${N}" "${TEST_CASE_TXT}" "${TYPE}" "${IMAGE}" "${NAMESPACE}" "${OUTPUT}" "${VALIDITY}"
+    N=$(($N+1))
+  done < "$1"
+}
+
+### INSTALLING CONNAISSEUR ####################################
+install() {
+  echo -n "Installing Connaisseur..."
+  # "make" is chosen deliberately below to test the Makefile, while other tests use "helm" directly
+  make install > /dev/null || { echo "${FAILED}"; exit 1; }
+  echo -e "${SUCCESS}"
+}
+
+upgrade() {
+  echo -n 'Upgrading Connaisseur...'
+  make upgrade > /dev/null || { echo -e ${FAILED}; exit 1; }
+  echo -e "${SUCCESS}"
+}
+
+### UNINSTALING CONNAISSEUR ####################################
+uninstall() {
+  echo -n 'Uninstalling Connaisseur...'
+  # 'make' is chosen deliberately below to test the Makefile, while other tests use 'helm' directly
```

Comment:
This probably makes no sense here. Also if this is intended to replace all, make sure we still test both make and helm

Reply:
added functions for helm as well